From d996a1bb4db84727fbf1a8e9461a032e04e544e7 Mon Sep 17 00:00:00 2001
From: Karel Gardas <karel.gardas@centrum.cz>
Date: Fri, 18 Jul 2014 23:54:52 -0500
Subject: [PATCH] fix inconsistency in exported functions from
 TcSplice.lhs/lhs-boot files when GHCI is not defined

Summary:
This patch fixes inconsistency in exported functions from TcSplice.lhs and TcSplice.lhs-boot
files. It looks like only GHC HEAD is sensitive to it and complains about it while bootstraping
another HEAD. At least this is what happening on Solaris/AMD64 builder machine where GHC 7.9.20140620
is used as a boostrap compiler. The failure does not happen on another builders.

Test Plan: validate

Reviewers: austin

Reviewed By: austin

Subscribers: phaskell, simonmar, relrod, carter

Differential Revision: https://phabricator.haskell.org/D74
---
 compiler/typecheck/TcSplice.lhs-boot | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

Index: ghc-7.8.3.20141119/compiler/typecheck/TcSplice.lhs-boot
===================================================================
--- ghc-7.8.3.20141119.orig/compiler/typecheck/TcSplice.lhs-boot	2014-12-08 18:49:28.251172880 +0100
+++ ghc-7.8.3.20141119/compiler/typecheck/TcSplice.lhs-boot	2014-12-08 18:49:28.251172880 +0100
@@ -3,7 +3,6 @@
 import HsSyn    ( HsSplice, HsBracket, HsQuasiQuote,
                   HsExpr, LHsType, LHsExpr, LPat, LHsDecl )
 import HsExpr   ( PendingRnSplice )
-import Id       ( Id )
 import Name     ( Name )
 import RdrName  ( RdrName )
 import TcRnTypes( TcM, TcId )
@@ -11,6 +10,7 @@
 import Annotations ( Annotation, CoreAnnTarget )
 
 #ifdef GHCI
+import Id       ( Id )
 import qualified Language.Haskell.TH as TH
 #endif
 
@@ -26,20 +26,20 @@
                -> TcRhoType
                -> TcM (HsExpr TcId)
 
-tcTopSpliceExpr :: Bool -> TcM (LHsExpr Id) -> TcM (LHsExpr Id)
-
 runQuasiQuoteDecl :: HsQuasiQuote RdrName -> TcM [LHsDecl RdrName]
 runQuasiQuoteExpr :: HsQuasiQuote RdrName -> TcM (LHsExpr RdrName)
 runQuasiQuoteType :: HsQuasiQuote RdrName -> TcM (LHsType RdrName)
 runQuasiQuotePat  :: HsQuasiQuote RdrName -> TcM (LPat RdrName)
 runAnnotation     :: CoreAnnTarget -> LHsExpr Name -> TcM Annotation
 
+#ifdef GHCI
+tcTopSpliceExpr :: Bool -> TcM (LHsExpr Id) -> TcM (LHsExpr Id)
+
 runMetaE :: LHsExpr Id -> TcM (LHsExpr RdrName)
 runMetaP :: LHsExpr Id -> TcM (LPat RdrName)
 runMetaT :: LHsExpr Id  -> TcM (LHsType RdrName)
 runMetaD :: LHsExpr Id -> TcM [LHsDecl RdrName]
 
-#ifdef GHCI
 lookupThName_maybe :: TH.Name -> TcM (Maybe Name)
 runQuasi :: TH.Q a -> TcM a
 #endif
